src/test/java/com/example/board/controller/PostControllerTest.java

```diff
@@ -0,0 +1,383 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureRestDocs
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@Transactional
+class PostControllerTest {
+
+    @Autowired
+    PostController postController;
+    @Autowired
+    ObjectMapper objectMapper;
+    @Autowired
+    PostService postService;
+    @Autowired
+    MockMvc mockMvc;
+
+
+    @Nested
+    @DisplayName("게시글 등록")
+    class PostNew {
+        @Test
+        @DisplayName("정상적으로 게시글을 등록된다.")
+        void saveSuccess() throws Exception {
+            //given
+            PostDto postDto = new PostDto(1L, "test2", "test Contents2");
+
+            //when, then
+            mockMvc.perform(post("http://localhost:8080/api/v1/posts")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(objectMapper.writeValueAsBytes(postDto)))
+                    .andExpect(status().isOk())
+                    .andDo(document("post-save",
+                            requestFields(
+                                    fieldWithPath("userId").type(JsonFieldType.NUMBER).description("userId"),
+                                    fieldWithPath("title").type(JsonFieldType.STRING).description("title"),
+                                    fieldWithPath("contents").type(JsonFieldType.STRING).description("contents")
+                            ),
+                            responseFields(
+                                    fieldWithPath("code").type(JsonFieldType.NUMBER).description("statusCode"),
+                                    fieldWithPath("data").type(JsonFieldType.NUMBER).description("postId"),
+                                    fieldWithPath("isSuccess").type(JsonFieldType.STRING).description("isSuccess")
+                            )
+                    ));
+        }
+
+        @Test
+        @DisplayName("userId를 포함하지 않고 보낼 경우 예외가 발생한다.")
+        void saveFailWithAnonymousUser() throws Exception {
+            //given
+            PostDto postDto = new PostDto(null, "test2", "test Contents2");
+
+            //when, then
+            String response = mockMvc.perform(post("http://localhost:8080/api/v1/posts")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(objectMapper.writeValueAsBytes(postDto)))
+                    .andExpect(status().isBadRequest())
+                    .andDo(print())
+                    .andReturn()
+                    .getResponse()
+                    .getContentAsString(StandardCharsets.UTF_8);
+            Response<List<String>> exception = objectMapper.readValue(response, Response.class);
+            assertThat(exception.getData().get(0)).isEqualTo("사용자 id를 입력해주세요.");
```

Comment:
이런 식으로 변경하시는 것은 어떻게 생각하시나요?
```suggestion
            assertThat(exception.getData()).containsOnly("사용자 id를 입력해주세요.");
```

Reply:
오오,, 하나 배워갑니다!